include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -806,14 +806,16 @@ struct __red_by_seg_op
     operator()(const _Tup1& __lhs_tup, const _Tup2& __rhs_tup) const
```

Comment:
Is it possible to fix the disambiguation with a trailing return type instead of the cast in the second return?
```suggestion
    operator()(const _Tup1& __lhs_tup, const _Tup2& __rhs_tup) const -> decltype(oneapi::dpl::__internal::make_tuple(get<0>(__lhs_tup), __binary_op(get<1>(__lhs_tup), get<1>(__rhs_tup)))
```

I'm not sure if that's better (it's arguably less readable). Just wondering if that would have any difference in terms of copies made as I do not know when the cast to `_OpOutType` would introduce a copy with these changes.

Reply:
I think that the number of copies is likely the same for either option, but I may be mistaken. (perhaps this one is worse because it may require copying the whole tuple, I'm not sure...)

Reply:
I wrote a simple example to test this with the conversion operator from our internal tuple to `std::tuple`. The trailing return type and the cast both result in a single call to this operator so either option works the same.

I think the cast is slightly more readable, so I chose this one with the updated name.